lib/Sema/CSDiag.cpp

```diff
@@ -5421,8 +5431,92 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
     // Type check the function subexpression to resolve a type for it if
     // possible.
     fnExpr = typeCheckChildIndependently(callExpr->getFn());
-    if (!fnExpr)
+    if (!fnExpr) {
+      auto UDE = dyn_cast<UnresolvedDotExpr>(callExpr->getFn());
+      if (!UDE)
+        return true;
```

Comment:
See my comment below about usage of `true`, I understand that original code used to do that but we should be more careful about it...